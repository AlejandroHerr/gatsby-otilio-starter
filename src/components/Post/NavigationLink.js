// @flow
import React from 'react';
import classnames from 'classnames';
import Link from 'gatsby-link';

import type { PreviewType } from '../../types/post';

import ByteIcon from '../icons/ByteIcon';

import styles from './Post.module.scss';

type PropsType = {
  className: string,
  icon: string,
} & PreviewType;

const NavigationLink = ({
  className, icon, excerpt, path, title,
}: PropsType) => (
  <Link
    className={classnames(styles.post_navigation__link, className)}
    to={path}
  >
    <section className={styles.post_navigation__teaser}>
      <div className={styles.post_navigation__icon}>
        <ByteIcon icon={icon} />
      </div>
      <h2 className={styles.post_navigation__title}>{title}</h2>
      <p className={styles.post_navigation__excerpt}>{excerpt}</p>
    </section>
  </Link>
);

export default NavigationLink;
