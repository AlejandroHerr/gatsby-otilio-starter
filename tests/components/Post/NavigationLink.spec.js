import React from 'react';
import renderer from 'react-test-renderer';

import NavigationLink from '../../../src/components/Post/NavigationLink';

const defaultProps = {
  className: 'testClassName',
  icon: 'testIcon',
  excerpt: 'testExcerpt',
  path: 'testPath',
  title: 'testTitle',
};

describe('Post', () => {
  describe('NavigationLink', () => {
    it('should match the exact snapshot', () => {
      const tree = renderer.create(<NavigationLink {...defaultProps} />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
