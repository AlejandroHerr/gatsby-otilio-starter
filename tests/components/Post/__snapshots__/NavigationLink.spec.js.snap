// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post NavigationLink should render a NavigationLink with a Gatsby Link 1`] = `
<GatsbyLink
  className="post_navigation__link testClassName"
  to="testPath"
>
  <section
    className="post_navigation__teaser"
  >
    <div
      className="post_navigation__icon"
    >
      <ByteIcon
        bold={false}
        className=""
        icon="testIcon"
      />
    </div>
    <h2
      className="post_navigation__title"
    >
      testTitle
    </h2>
    <p
      className="post_navigation__excerpt"
    >
      testExcerpt
    </p>
  </section>
</GatsbyLink>
`;
